src/portage_util__whirlpool.c

```diff
@@ -1112,30 +1112,42 @@ static PyTypeObject WhirlpoolType = {
     .tp_methods = Whirlpool_methods,
 };
 
+static int
+whirlpool_exec(PyObject *mod)
+{
+    if (PyType_Ready(&WhirlpoolType) < 0)
+        return -1;
+
+    Py_INCREF(&WhirlpoolType);
+    if (PyModule_AddObject(mod, "Whirlpool", (PyObject *) &WhirlpoolType) < 0) {
```

Comment:
This passes tests but I'm not sure if it's really correct. I've seen other modules like https://github.com/python/cpython/blob/main/Modules/termios.c store all of their state in a struct.

Reply:
The docs here indicate that state should be specific to the module object (e.g. using PyModule_GetState()):

https://docs.python.org/3.13/c-api/module.html#multi-phase-initialization

Reply:
I think this is okay because the "state" is WhirlpoolType an it's immutable.